File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/group/model/GroupVO.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.group.model;

import com.zfoo.protocol.IPacket;

import java.util.List;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2020-04-22 11:43
 */
public class GroupVO implements IPacket {

    public static final transient short PROTOCOL_ID = 18000;

    private long id;

    private long adminId;

    private List<ChannelBoxVO> channelBoxes;

    private String name;

    private String avatar;

    private String background;

    private long createTime;

    private List<GroupAuthVO> groupAuths;

    public static GroupVO valueOf(long id, long adminId, List<ChannelBoxVO> channelBoxes, String name, String avatar, String background
            , long createTime, List<GroupAuthVO> groupAuths) {
        var vo = new GroupVO();
        vo.id = id;
        vo.adminId = adminId;
        vo.channelBoxes = channelBoxes;
        vo.name = name;
        vo.avatar = avatar;
        vo.background = background;
        vo.createTime = createTime;
        vo.groupAuths = groupAuths;
        return vo;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getAdminId() {
        return adminId;
    }

    public void setAdminId(long adminId) {
        this.adminId = adminId;
    }

    public List<ChannelBoxVO> getChannelBoxes() {
        return channelBoxes;
    }

    public void setChannelBoxes(List<ChannelBoxVO> channelBoxes) {
        this.channelBoxes = channelBoxes;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getBackground() {
        return background;
    }

    public void setBackground(String background) {
        this.background = background;
    }

    public long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public List<GroupAuthVO> getGroupAuths() {
        return groupAuths;
    }

    public void setGroupAuths(List<GroupAuthVO> groupAuths) {
        this.groupAuths = groupAuths;
    }
}

File: web/src/jsProtocol/group/model/GroupVO.js
// @author jaysunxiao
// @version 1.0
// @since 2020-04-22 11:43
const GroupVO = function(adminId, avatar, background, channelBoxes, createTime, groupAuths, id, name) {
    this.adminId = adminId; // long
    this.avatar = avatar; // java.lang.String
    this.background = background; // java.lang.String
    this.channelBoxes = channelBoxes; // java.util.List<com.zfoo.app.zapp.common.protocol.group.model.ChannelBoxVO>
    this.createTime = createTime; // long
    this.groupAuths = groupAuths; // java.util.List<com.zfoo.app.zapp.common.protocol.group.model.GroupAuthVO>
    this.id = id; // long
    this.name = name; // java.lang.String
};

GroupVO.prototype.protocolId = function() {
    return 18000;
};

GroupVO.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writeLong(packet.adminId);
    buffer.writeString(packet.avatar);
    buffer.writeString(packet.background);
    buffer.writePacketArray(packet.channelBoxes, 18011);
    buffer.writeLong(packet.createTime);
    buffer.writePacketArray(packet.groupAuths, 18001);
    buffer.writeLong(packet.id);
    buffer.writeString(packet.name);
};

GroupVO.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new GroupVO();
    const result0 = buffer.readLong();
    packet.adminId = result0;
    const result1 = buffer.readString();
    packet.avatar = result1;
    const result2 = buffer.readString();
    packet.background = result2;
    const list3 = buffer.readPacketArray(18011);
    packet.channelBoxes = list3;
    const result4 = buffer.readLong();
    packet.createTime = result4;
    const list5 = buffer.readPacketArray(18001);
    packet.groupAuths = list5;
    const result6 = buffer.readLong();
    packet.id = result6;
    const result7 = buffer.readString();
    packet.name = result7;
    return packet;
};

export default GroupVO;
